File: eTheater.Model/ShowActor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eTheater.Model
{
    public class ShowActor
    {
        public int ShowActorId { get; set; }

        public int? ActorId { get; set; }

        public int? ShowId { get; set; }

        public virtual Actor? Actor { get; set; }

        public virtual Show? Show { get; set; }

        public bool IsDeleted { get; set; }
    }
}

File: UI/etheater_web/lib/models/showActor.dart
import 'package:etheater_web/models/models.dart';
import 'package:json_annotation/json_annotation.dart';

part 'showActor.g.dart';

@JsonSerializable()
class ShowActor {
  int showActorId;
  int actorId;
  int showId;
  Actor? actor;
  Show? show;
  bool? isDeleted;

  ShowActor(
      {required this.showActorId,
      required this.actorId,
      required this.showId,
      this.actor,
      this.show,
      this.isDeleted});

  factory ShowActor.fromJson(Map<String, dynamic> json) =>
      _$ShowActorFromJson(json);
  Map<String, dynamic> toJson() => _$ShowActorToJson(this);
}
